core/src/replay_stage.rs

```diff
@@ -3344,7 +3344,7 @@ impl ReplayStage {
                 SwitchForkDecision::FailedSwitchThreshold(switch_proof_stake, total_stake) => {
                     let reset_bank = heaviest_bank_on_same_voted_fork;
                     let final_switch_fork_decision = Self::select_forks_failed_switch_threshold(
-                        reset_bank,
+                        reset_bank.map(|bank| bank.as_ref()),
```

Comment:
This change is necessary, altho it might not have looked like it due to my intermediate commits not compiling. `heaviest_bank` is an `&Arc<Bank>`, but we need to supply a `&Bank` to the subfunction:
```
error[E0308]: mismatched types
    --> core/src/replay_stage.rs:3348:25
     |
3347 |                     let final_switch_fork_decision = Self::select_forks_failed_switch_threshold(
     |                                                      ------------------------------------------ arguments to this function are incorrect
3348 |                         reset_bank,
     |                         ^^^^^^^^^^ expected `Option<&Bank>`, found `Option<&Arc<Bank>>`
     |
     = note: expected enum `std::option::Option<&solana_runtime::bank::Bank>`
                found enum `std::option::Option<&Arc<solana_runtime::bank::Bank>>`
 ```